core/FrontController.php

```diff
@@ -133,6 +148,15 @@ public function dispatch($module = null, $action = null, $parameters = null)
              * @param \Piwik\NoAccessException $exception The exception that was caught.
              */
             Piwik::postEvent('User.isNotAuthorized', array($exception), $pending = true);
+        } catch (\Twig_Error_Runtime $e) {
+            echo $this->generateSafeModeOutputFromException($e);
+            exit;
+        } catch(StylesheetLessCompileException $e) {
+            echo $this->generateSafeModeOutputFromException($e);
+            exit;
+        } catch(\Error $e) {
+            echo $this->generateSafeModeOutputFromException($e);
```

Comment:
Here we're catching errors such as: #10876 `Call to undefined method Piwik\Menu\MenuTop::add()` which should be displayed safe mode instead of a not-so-useful error page